kong/db/schema/init.lua

```diff
@@ -1599,6 +1610,33 @@ function Schema:validate(input, full_check)
 end
 
 
+-- Iterate through input fields on update and check agianst schema for 
+-- immutable attribute. If immutable attribute is set, compare input values
+-- against entity values to detirmine whether input is valid.
+-- @param input The input table.
+-- @param entity The entity update will be performed on.
+-- @return True on success.
+-- On failure, it returns nil and a table containing all errors by field name.
+-- In all cases, the input table is untouched.
+function Schema:validate_immutable_fields(input, entity)
+  local errors = {}
+
+  input = tablex.deepcopy(input)
+
+  for key, field in pairs(input) do
```

Comment:
Have you considered using `each_field()` to iterate over this input table, as the more native way to iterate over fields of a schema-based entity?

```lua
for name, field in self:each_field() do
  local value = input[name]
  if field.immutable and ... then

  end
end
```

Reply:
👍 good call